File: Odd Even Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        
        if(head==NULL || head->next==NULL)
            return head;
        ListNode* oddend=head;
        ListNode* evenstart=head->next;
        ListNode* tmp=head;
        ListNode* crawlp=evenstart;
        while(crawlp!=NULL && crawlp->next!=NULL )
        {
            ListNode* crawl=crawlp->next;
            crawlp->next=crawl->next;
            oddend->next=crawl;
            crawl->next=evenstart;
            oddend=oddend->next;
            crawlp=crawlp->next;
            
        }
        head=tmp;
        return head;
        
    }
};

File: Golang Solutions/Odd Even Linked List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func oddEvenList(head *ListNode) *ListNode {
    if head==nil || head.Next==nil{
        return head
    }
    oddend:=head
    evenstart:=head.Next
    tmp:=head
    crawlp:=evenstart
    for ;crawlp!=nil && crawlp.Next!= nil;{
        crawl:=crawlp.Next
        crawlp.Next=crawl.Next
        oddend.Next=crawl
        crawl.Next=evenstart
        oddend=oddend.Next
        crawlp=crawlp.Next
    }
    head=tmp
    return head
    
}